stage1/init/init.go

```diff
@@ -171,6 +172,50 @@ func machinedRegister() bool {
 	return found == 2
 }
 
+func lookupPath(bin string, paths string) (string, error) {
+	pathsArr := strings.Split(paths, ":")
+	for _, path := range pathsArr {
+		binPath := filepath.Join(path, bin)
+		binAbsPath, err := filepath.Abs(binPath)
+		if err != nil {
+			return "", fmt.Errorf("unable to find absolute path for %s", binPath)
+		}
+		if _, err := os.Stat(binAbsPath); err == nil {
+			return binAbsPath, nil
+		}
+	}
+	return "", fmt.Errorf("unable to find %q in %q", bin, paths)
+}
+
+func installAssets() error {
+	systemctlBin, err := lookupPath("systemctl", os.Getenv("PATH"))
+	if err != nil {
+		return err
+	}
+	bashBin, err := lookupPath("bash", os.Getenv("PATH"))
+	if err != nil {
+		return err
+	}
+	systemdShutdownBin, err := lookupPath("systemd-shutdown", "/lib/systemd:/usr/lib/systemd")
```

Comment:
On my Debian, `systemd-shutdown` is in `/lib/systemd/`. On @iaguis' Arch, `systemd-shutdown` is `/usr/lib/systemd/`. I'm not sure what do do in the general case. Systemd has:
```
Makefile.am:192:        -DSYSTEMD_SHUTDOWN_BINARY_PATH=\"$(rootlibexecdir)/systemd-shutdown\" \
```

But this `-D` define does not seem to be exported in a symbol.